### PR TITLE
feat(logout_uri): add post logout dynamic url for external authenticat…

### PR DESCRIPTION
**Pull request by LucassvTI**
## After this PR
Added post logout dynamic url for external authenticator redirection and corrected query separator parameters of ? for &

### CODE REVIEW

File: packages/react/src/oidc/vanilla/oidc.ts [hunk 1]
```diff
@@ -628,24 +628,29 @@ Please checkout that you are using OIDC hook inside a <OidcProvider configuratio
          timer.clearTimeout(this.timeoutId);
      }
      
-    async logoutAsync(callbackPath: string | undefined = undefined, extras:StringMap=null) {
+    async logoutAsync(callbackPath: string | undefined = undefined, extras:StringMap=null, postLogoutRedirectUri: string | undefined = undefined) {
```
Comment (guillaume-chervet) [inline packages/react/src/oidc/vanilla/oidc.ts:631]: I would prefer something like that.

`
 async logoutAsync(callbackPathOrUrl: string | undefined = undefined, extras:StringMap=null) {
        const configuration = this.configuration;
        const oidcServerConfiguration = await this.initAsync(configuration.authority, configuration.authority_configuration);
        if(callbackPath && (typeof callbackPath !== 'string'))
        {
            callbackPath = undefined;
            console.warn('callbackPath path is not a string');
        }
        const path = (callbackPath === null || callbackPath === undefined) ? location.pathname + (location.search || '') + (location.hash || '') : callbackPath;
		const isUri = callbackPathOrUrl.includes("https://") || callbackPathOrUrl.includes("http://");
		let url = isUri ? callbackPathOrUrl : window.location.origin + path;
        // @ts-ignore
        const idToken = this.tokens ? this.tokens.idToken : "";
        await this.destroyAsync();  
        if(oidcServerConfiguration.endSessionEndpoint) {
            let extraQueryString = "";
            if(extras){
                for (let [key, value] of Object.entries(extras)) {
                    extraQueryString +=`&${key}=${encodeURIComponent(value)}`;
                }
            }
            window.location.href = `${oidcServerConfiguration.endSessionEndpoint}?post_logout_redirect_uri=${encodeURIComponent(url)}&id_token_hint=${idToken}${extraQueryString}`;
}
`

**Comment by guillaume-chervet** [comment]
thank you very much @LucassvTI this is awesome.
I made you a feeback

File: packages/react/src/oidc/vanilla/oidc.ts (hunk 1 above)
Comment (guillaume-chervet) [reply packages/react/src/oidc/vanilla/oidc.ts:631]: I prefer that solution. I found i cleaner and simpler to use

Comment (LucassvTI) [reply packages/react/src/oidc/vanilla/oidc.ts:631]: I like this solution.

**Review by guillaume-chervet — APPROVED** [review]

File: packages/react/src/oidc/vanilla/oidc.ts (hunk 1 above)
Comment (guillaume-chervet) [reply packages/react/src/oidc/vanilla/oidc.ts]: Thank you very much @LucassvTI. 
Awesome!

**Pull request merged**